crates/ide/src/inlay_hints/param_name/generic.rs

```diff
@@ -0,0 +1,288 @@
+//! Implementation of inlay hints for generic parameters.
+use ide_db::{active_parameter::generic_def_for_node, RootDatabase};
+use syntax::{
+    ast::{self, HasName},
+    AstNode, SyntaxNode,
+};
+
+use crate::{inlay_hints::GenericParameterHints, InlayHint, InlayHintsConfig, InlayKind};
+
+use super::{is_argument_similar_to_param_name, render_label};
+
+pub(crate) fn hints(
+    acc: &mut Vec<InlayHint>,
+    sema: &hir::Semantics<'_, RootDatabase>,
+    config: &InlayHintsConfig,
+    node: SyntaxNode,
```

Comment:
```suggestion
    node: AnyHasGenericArgs,
```
[`c08d419` (#17555)](https://github.com/rust-lang/rust-analyzer/pull/17555/commits/c08d419fba1c7ad3c6e80f38d9412942c8516b9a) adds that, we should use it here (will need a rebase)